### PR TITLE
Add a CI workflow using GitHub Actions

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | ~no~ probably!
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Comment by teohhanhui** [comment]
I can't get the checks to be triggered in the pull request...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: CI
+
+on: push
```
Comment (bendavies) [inline .github/workflows/ci.yml]: `on: [push, pull_request]`?

Comment (teohhanhui) [reply .github/workflows/ci.yml]: The thing is:

> If you intend to use the pull_request event to trigger CI tests, we recommend that you set up your workflow configuration to listen to the push event.

https://help.github.com/en/articles/events-that-trigger-workflows#pull-request-events-for-forked-repositories

I'm not sure what that means, and not sure if it matches the latest implementation.

Comment (bendavies) [reply .github/workflows/ci.yml]: yeah i just read that too. very confusing

Comment (bendavies) [reply .github/workflows/ci.yml]: you see it ran in your fork?
https://github.com/teohhanhui/api-platform-core/runs/209804563

Comment (teohhanhui) [reply .github/workflows/ci.yml]: Yeah, I know, that's why I'm thinking of merging this into `api-platform/core` and see how it goes.

Comment (teohhanhui) [reply .github/workflows/ci.yml]: Found this: https://stackoverflow.com/a/57621843/1529493

Comment (teohhanhui) [reply .github/workflows/ci.yml]: I've updated accordingly. Let's see whether it triggers on PRs correctly after this is merged.

**Comment by teohhanhui** [comment]
See https://github.com/teohhanhui/api-platform-core/commit/ef7ffce940fb800b9d2fce60ebde979574f5fd7c/checks for the current workflow status.

**Comment by teohhanhui** [comment]
Yes, it's being triggered correctly! :rocket: See https://github.com/api-platform/core/pull/3021/checks

**Pull request merged**